analyze/aps-analyze.c

```diff
@@ -17,35 +17,43 @@ Several phases:
 static void *analyze_thing(void *ignore, void *node)
 {
   STATE *s;
+  DEPENDENCY d;
   if (ABSTRACT_APS_tnode_phylum(node) == KEYDeclaration) {
     Declaration decl = (Declaration)node;
     switch (Declaration_KEY(decl)) {
     case KEYmodule_decl:
       s = compute_dnc(decl);
-      switch (analysis_state_cycle(s)) {
+      d = analysis_state_cycle(s);
+      switch (d) {
       default:
-	aps_error(decl,"Cycle detected; Attribute grammar is not DNC");
+	aps_error(decl,"Cycle detected; Attribute grammar is not DNC %d", d);
 	break;
+      case indirect_dependency:
```

Comment:
It seems wrong to me to have "indirect_dependency" here.  These are possibly simple dependencies.